src/main/scala/org/hammerlab/guacamole/alignment/AffineGapPenaltyAlignment.scala

```diff
@@ -0,0 +1,157 @@
+package org.hammerlab.guacamole.alignment
+
+import breeze.linalg.{ DenseMatrix, argmin }
+import org.hammerlab.guacamole.alignment.AlignmentState.AlignmentState
+
+import scala.annotation.tailrec
+
+object AffineGapPenaltyAlignment {
+
+  /**
+   * Retraces the optimal alignment path from a matrix of scores and transitions
+   *
+   * @param alignmentStates sequenceLength x referenceLength matrix of alignment score at each step
+   * @param alignmentScores sequenceLength x referenceLength matrix of best transition at each step
+   * @param sequenceLength Length of the input sequence to align
+   * @param referenceLength Length of reference sequence to align against
+   * @return An alignment path and score
+   */
+  def findBestPath(alignmentStates: DenseMatrix[AlignmentState],
+                   alignmentScores: DenseMatrix[Double],
+                   sequenceLength: Int,
+                   referenceLength: Int): ReadAlignment = {
+
+    @tailrec
+    def recFindBestPath(sequenceIdx: Int,
+                        referenceIdx: Int,
+                        currentPath: List[AlignmentState]): List[AlignmentState] = {
+
+      val lastMove = alignmentStates(sequenceIdx, referenceIdx)
```

Comment:
maybe a basic breeze question, but doesn't this look for an invalid row and column the first time this is called, at `(sequenceLength, referenceLength)`?